telemetry/telemetry.go

```diff
@@ -0,0 +1,111 @@
+// As part of the White Glove White Glove project,
+// this is a telemetry package to support posting summary and cancellation events
+// to a given webhook endpoint with a reference project id in the shape of an event grid
+// custom event.
+
+package telemetry
+
+import (
+	"bytes"
+	"encoding/json"
+	"log"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/Azure/azure-storage-azcopy/common"
+)
+
+var (
+	// ReportingAPIURI the Webhook URI
+	ReportingAPIURI string
+	// ProjectReferenceID the project reference ID
+	ProjectReferenceID string
+)
+
+type telemetry struct {
+	JobID     string      `json:"jobID"`
+	ProjectID string      `json:"projectId"`
+	Telemetry interface{} `json:"telemetry"`
+}
+
+type reportEvent struct {
+	EventType  string      `json:"eventType"`
+	EventTopic string      `json:"eventTopic"`
+	Data       interface{} `json:"data"`
+	EventTime  time.Time   `json:"eventTime"`
+}
+
+func getJSON(t interface{}) string {
+	json, err := json.Marshal(t)
+	common.PanicIfErr(err)
+	return string(json)
```

Comment:
I don't know how feasible it is for something to fail json marshalling, but, please don't panic if it fails to marshal.

Reply:
IIRC the usual approach with logging is to just ignore errors.  (Since you can't exactly _log_ them when its logging that is failing).  By ignore, I mean, if we can't log _this_ telemetry/reportEvent, then skip it, and just be ready to log the next one.